tests/smartswitch/common/device_utils_dpu.py

```diff
@@ -0,0 +1,164 @@
+"""
+Helper script for DPU  operations
+"""
+import logging
+import pytest
+from tests.common.devices.sonic import *  # noqa: F401,F403
+from tests.platform_tests.api.conftest import *  # noqa: F401,F403
+from tests.common.helpers.platform_api import chassis, module
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+from pkg_resources import parse_version
+
+
+@pytest.fixture(scope='function')
+def num_dpu_modules(platform_api_conn):
+
+    """
+    Returns the number of DPU modules
+    """
+
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+    return num_modules
+
+
+@pytest.fixture(scope='function')
+def skip_test_smartswitch(duthosts, enum_rand_one_per_hwsku_hostname,
+                          platform_api_conn):
+    """
+    Checks whether given testbed is running
+    202405 image or below versions
+    If True, then skip the script
+    else checks if dpus are in darkmode
+    If dpus are in dark mode, then power up the DPUs
+    else, proceeds to run all test cases
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    if not duthost.facts["DPUS"] and \
+            parse_version(duthost.os_version) <= parse_version("202405"):
+        pytest.skip("Test is not supported for this testbed and os version")
+
+    darkmode = is_dark_mode_enabled(duthost, platform_api_conn)
+
+    if darkmode:
+        dpu_power_on(duthost, platform_api_conn)
+
+
+def is_dark_mode_enabled(duthost, platform_api_conn):
+
+    """
+    Checks the liveliness of DPU
+    Returns:
+        True if all DPUs admin status are down
+        else False
+    """
+
+    num_modules = num_dpu_modules(platform_api_conn)
+    count_admin_down = 0
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        output_config_db = duthost.command(
+                           'redis-cli -p 6379 -h 127.0.0.1 \
+                            -n 4 hgetall "CHASSIS_MODULE|{}"'.format(dpu))
+        if 'down' in output_config_db['stdout']:
+            count_admin_down += 1
+
+    if count_admin_down == num_modules:
+        return True
+
+    return False
+
+
+def dpu_power_on(duthost, platform_api_conn, check_dpu_ping_status):
+    """
+    Executes power on all DPUs
+    Returns:
+        Returns True or False based on all DPUs powered on or not
+    """
+
+    num_modules = num_dpu_modules(platform_api_conn)
+    ip_address_list = []
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        ip_address_list.append(
+                module.get_midplane_ip(platform_api_conn, index))
+        duthost.shell("config chassis modules startup %s" % (dpu))
+
+    pytest_assert(wait_until(180, 60, 0, check_dpu_ping_status,
+                  duthost, ip_address_list), "Not all DPUs operationally up")
```

Comment:
Minor nit: Use "Not all DPUs are operationally up"

Reply:
Changed it.